echo_server.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import print_function
+import socket
+import os
+
+
+s = socket.socketpair(socket.AF_UNIX, socket.SOCK_STREAM)
+print ('Socket created')
+
+
+def socketpair():
+    done = False
+    buffer_size = 16
+    response = ''
+    try:
+        parent, child = s
+        pid = os.fork()
+        print('PID = ', pid)
+        if pid:
+            parent.close()
+            child.send("Hello parent, do you hear me?")
+            while not done:
+                child_message = child.recv(buffer_size)
+                if len(child_message) < buffer_size:
+                    done = True
+                response += child_message
+            print("parent sent '%s'" % response)
+        else:
+            child.close()
+            parent.send("Yes child, I hear you.")
+            while not done:
+                parent_message = parent.recv(buffer_size)
+                if len(parent_message) < buffer_size:
+                    done = True
+                response += parent_message
+            print("child sent '%s'" % response)
+    except:
+        raise
+
+if __name__ == '__main__':
+    socketpair()
```

Comment:
While this works for the simple problem of sending the one message to and receiving the other message back, it doesn't really solve the problem of having an echo server and a client to send arbitrary messages to it.  

The problem is that the socket pair is really not a server/client relationship.  It's a peer-to-peer relationship.  Neither is the server in this case.

Can you re-write these scripts such that you have a real server and a separate client?  I think it'll help you to understand the way that HTTP communications work, which is really the end goal of the exercise.